### PR TITLE
Add xhtml to list of valid extensions

### PR DESCRIPTION
**Pull request by waldyrious**
This also adds the `application/xhtml+xml` mime type, which is recommended<sup>[1][2][3]</sup> and widely supported<sup>[4]</sup>, but not yet returned by magic/file<sup>[5][6]</sup> for `*.xhtml` files.

This means that `Site.valid_file_type?` does not need the mime type to be in the list, but should [libmagic/file](http://www.darwinsys.com/file/) ever adopt W3C's recommendation, Neocities will be ready for that.

[1] https://www.w3.org/International/articles/serving-xhtml/#mime
[2] https://www.w3.org/2003/01/xhtml-mimetype/
[3] https://www.w3.org/TR/xhtml-media-types/#media-types
[4] https://caniuse.com/xhtml
[5] https://github.com/qoobaa/magic/blob/a93a0fac5/lib/magic.rb#L36-L38
[6] https://github.com/file/file/blob/54393977c/magic/Magdir/sgml#L28-L40

### CODE REVIEW

**Comment by waldyrious** [comment]
Hey @kyledrake, just wondering if you could take a look at this sometime. I just bumped into this again :)

**Comment by jeremyredhead** [comment]
You might also want to add .xht (shorter form of .xhtml) to that list (as obscure as it is).

**Comment by waldyrious** [comment]
Thanks for the suggestion, @jeremyredhead! I'll do that along with the rebase to fix the git conflicts, once I get the OK from the maintainers :)

**Comment by lime360** [comment]
i'm doing it by myself

**Comment by kyledrake** [comment]
Can you explain why xhtml support should be added to the free sites? We generally want to support modern html5 sites, I really don't understand the use case here.

**Comment by kyledrake** [comment]
I'm also confused why this needs to be it's own extension and can't just be used as .html.

**Comment by jeremyredhead** [comment]
TL;DR: I'd love XHTML support for no particularly good reason but it may not be worth the hassle.

> Can you explain why xhtml support should be added to the free sites? We generally want to support modern html5 sites, I really don't understand the use case here.

Why not? Some people might just want the option to write XHTML instead of HTML. Aside from a higher likelihood of a completely broken page (XML parsing error) if someone doesn't know what they're doing, there shouldn't be any major differences between XHTML and HTML I don't think.

> I'm also confused why this needs to be its own extension and can't just be used as `.html`

Well, if XHTML is served as .html, it will be parsed as HTML. HTML parsing can be kinda goofy at times.
If XHTML is served as XML, or XHTML, it will be parsed as XML -- which is much stricter & also simpler syntactically. Some people may prefer that strictness. Some people might want to use XML features that HTML doesn't let them (e.g. self-closing tags)

That said, if one really wants XML parsing, they should currently be able to just serve their XHTML as .xml files and (assuming they've set the xmlns correctly) it should just work. Especially now that Neocities aggressively hides file extensions (for better or worse). And for that matter... I'm already unsure which wins out if both a .htm and a .html file exist with the same path/name... Adding .xhtml/.xht into the mix there may be inadvisable and confusing wrt "clean" filepaths. So as much as it would delight & bemuse me to see .xhtml/.xht support, it might be better to leave it out. (P.S. I *pinky promise* despite the rambling-ness of this response that I didn't use ChatGPT... I just write like this `-_-;`)

**Comment by kyledrake** [comment]
Github deleted my previous comment for some reason, but TLDR I've decided to not merge this, as refactoring all of the .html stuff across the stack to support a new file extension is going to be very difficult, and an .html file can easily be set to xhtml by setting the correct doctype ar the top of the file, which is required for xhtml anyways https://www.w3schools.com/html/html_xhtml.asp
